### PR TITLE
_collect_type_vars should not collect Unpack objects itself

### PR DESCRIPTION
**Pull request by Daraan**
Fixes:  #471

- #471

This PR fixes that these two cases should only yield `T` and not the unpack variable itself:

```py
print(_collect_type_vars([Unpack[Movie[T]]]))
# (typing_extensions.Unpack[typing_extensions.Movie[~T]], ~T)

print(_collect_type_vars([Unpack[Tuple[T]]]))
# (typing_extensions.Unpack[typing.Tuple[~T]], ~T)
```

### CODE REVIEW

File: src/test_typing_extensions.py [hunk 1]
```diff
@@ -5779,6 +5779,33 @@ class D(Protocol[T1, T2, Unpack[Ts]]): pass
                     with self.assertRaises(TypeError):
                         klass[int]
 
+    def test_with_non_typevartuple(self):
+        # The two functions below are used by typing._GenericAlias and typing.Generic via monkey patching.
```
Comment (JelleZijlstra) [inline src/test_typing_extensions.py:5783]: Could you add a test that exercises public functionality? `_collect_type_vars` is a private function and it doesn't matter what it does; we're interested in something like `MyTypedDictT[int]` misbehaving.

It's fine to keep the tests you have here as more focused regression tests.

File: src/test_typing_extensions.py [hunk 2]
```diff
@@ -5805,6 +5806,11 @@ class MyTypedDictT(Generic[T], TypedDict):
             self.assertEqual(typing_extensions._collect_parameters((Unpack[MyTypedDictT[T]], )), (T,))
             self.assertEqual(typing_extensions._collect_parameters((Unpack[Tuple[int]], )), ())
             self.assertEqual(typing_extensions._collect_parameters((Unpack[Tuple[int, T]], )), (T,))
+        self.assertEqual(typing._GenericAlias(Any, Unpack[MyTypedDict]).__parameters__, ())
```
Comment (JelleZijlstra) [inline src/test_typing_extensions.py:5809]: This is still a private object that shouldn't be used directly. Can you focus the tests on use of public functionality that currently doesn't work and is fixed by your change?

Comment (Daraan) [reply src/test_typing_extensions.py:5809]: I think I then misunderstood what you meant by "keep the tests here" 🙃 . Can you clarify that for me?

I interpreted "here" as in the code, but it should have some other case added. Do I then interpret it correctly that you mean "here" with here in this PR and I should remove this whole block for the private function?

Comment (JelleZijlstra) [reply src/test_typing_extensions.py:5809]: Thinking about it more: Let's have this PR add only tests that exercise public functionality, and remove any tests that test only private functions.

**Comment by JelleZijlstra** [comment]
This is looking good now, could you also add a changelog entry like Alex added on a few other PRs today?

**Review by AlexWaygood — APPROVED** [review]
Thanks!

**Review by JelleZijlstra — APPROVED** [review]

**Comment by Daraan** [comment]
> Thanks!

Thanks to you two as well!

**Pull request merged**